### PR TITLE
Verify the PCIe daemon status before retrieving the PCIE_DEVICES table from the database

### PR DESCRIPTION
**Pull request by vvolam**
### Description of PR

Summary:
The "get_pcie_devices_tbl_key" fixture is currently invoked before the "setup" fixture. When the PCIe daemon is not running, the "get_pcie_devices_tbl_key" fixture asserts that the PCIE_DEVICES table is empty in the database and causes the test to fail. Instead, make the "get_pcie_devices_tbl_key" fixture dependent on the "setup" fixture to verify the daemon status before retrieving the PCIE_DEVICES table from the database, which skips the test if daemon is not running.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [x] 202012
- [x] 202205
- [x] 202305
- [x] 202311

#### What is the motivation for this PR?

There are occasional failures with the "PCIE_DEVICES table is empty" message when running PCIe tests. This fix ensures the PCIe daemon is running before retrieving the PCIE_DEVICES table, thereby increasing test reliability.

#### How did you do it?

Make the "get_pcie_devices_tbl_key" fixture dependent on the "setup" fixture to verify the daemon status before retrieving the PCIE_DEVICES table from the database.

#### How did you verify/test it?
- Ran test_pcied.py tests on lab testbeds.

### CODE REVIEW

File: tests/platform_tests/daemon/test_pcied.py [hunk 1]
```diff
@@ -52,6 +52,18 @@ def setup(duthosts, rand_one_dut_hostname):
         pytest.skip("{} is not enabled in {} {}".format(daemon_name, duthost.facts['platform'], duthost.os_version))
 
 
+@pytest.fixture(scope="module", autouse=True)
+def get_pcie_devices_tbl_key(duthosts, rand_one_dut_hostname, setup):
```
Comment (prgeor) [inline tests/platform_tests/daemon/test_pcied.py:56]: @vvolam `setup` argument doesn't seem to be used?

Comment (vvolam) [reply tests/platform_tests/daemon/test_pcied.py:56]: By including the setup() fixture as an argument, it is executed before the get_pcie_devices_tbl_key() fixture. This adjustment ensures that the setup() fixture runs first, confirming that the pcied daemon is active before proceeding.

**Comment by vvolam** [comment]
@prgeor @vaibhavhd @vdahiya12, please review.

**Review by vaibhavhd — APPROVED** [review]

**Review by vdahiya12 — APPROVED** [review]

**Review by mihirpat1 — APPROVED** [review]

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@vvolam i think we need this fix in other branches > 20230531. so need to cherry pick?

**Comment by vvolam** [comment]
@prgeor I have requested for 202305 and 202311 branches. Could you approve?

**Pull request merged**